File: API/sensors/sensors.py
from ADC import *
from I2C import *
from one_wire import *
from UART import *

# populate sensor suite by sensor category
sensor_suite = {'EC1': electrical_conductivity("electrical_conductivity",  0x66),
		'pH1':	       	    I2C_sensor("ph_and_circuitry",         0x65),
		'temp1':       	   temperature("liquid_temp",             "28-00000673a8a7"),
		'DO':          	    I2C_sensor("do_probe_and_circuitry",   0x61),
		'LL1': 	       	  liquid_level("liquid_level",             0x80),
		'LL2': 	       	  liquid_level("liquid_level",             0xA0),
		'LL3': 	       	  liquid_level("liquid_level",             0xC0),
		'LL4': 	       	  liquid_level("liquid_level",             0xE0),
		'LL5': 	       	  liquid_level("liquid_level",             0xD0),
		'flow_meter1': 	    flow_meter("flow_meter_and_circuitry", 1),
		'flow_meter2': 	    flow_meter("flow_meter_and_circuitry", 4),
		'LL6': 	          liquid_level("liquid_level",             0xF0),
		'temp2':           temperature("soil_temp",               "28-0000065f27cc"),
		'temp3':           temperature("soil_temp",               "28-0000065eb57a"),
		'temp4':           temperature("soil_temp",               "28-000006747f7f"),
		'EC2': electrical_conductivity("electrical_conductivity",  0x64),
		'pH2':	            I2C_sensor("ph_and_circuitry",	   0x63),
		'MO1':	              moisture("moisture",                 0x80),
		'MO2':	              moisture("moisture",                 0xA0),
		'MO3':	              moisture("moisture",                 0xC0),
		'MO4':	              moisture("moisture",                 0xE0),
		'RHT1':	           RH_and_temp("rh_and_air_temp",         "P8_8"),
		'RHT2':	           RH_and_temp("rh_and_air_temp",         "P8_9"),
		'TP1':	        total_pressure("total_pressure",           2),
		'O2':	                oxygen("O2"),
		'CO2':	                   CO2("CO2",                      5),
		'PAR1':	                   PAR("light",                    0xF0),
		'RHT3':	           RH_and_temp("rh_and_air_temp",         "P8_10"),
		'TP2':	        total_pressure("total_pressure",           1),
		'PAR2':                    PAR("light",                    0xD0)}

# populate sensor suite by SysID
S = {
     # liquid tanks and plumbing
     101: electrical_conductivity("electrical_conductivity",  0x66),
     102:              I2C_sensor("ph_and_circuitry",         0x65),
     103:             temperature("liquid_temp",             "28-00000673a8a7"),
     104:              I2C_sensor("do_probe_and_circuitry",   0x61),
     105:            liquid_level("liquid_level", 	      0x80), 
     106:            liquid_level("liquid_level", 	      0xA0),
     107:            liquid_level("liquid_level", 	      0xC0),
     108:            liquid_level("liquid_level", 	      0xE0),
     109:            liquid_level("liquid_level", 	      0xD0),
     110:              flow_meter("flow_meter_and_circuitry", 1),
     111:              flow_meter("flow_meter_and_circuitry", 4),
     112:            liquid_level("liquid_level", 	      0xF0),

     # growth medium
     201: 	       temperature("soil_temp", 	      "28-0000065f27cc"), 
     202: 	       temperature("soil_temp", 	      "28-0000065eb57a"),
     203: 	       temperature("soil_temp", 	      "28-000006747f7f"),
     205:  electrical_conductivity("electrical_conductivity",  0x64),
     206:  		I2C_sensor("ph_and_circuitry",	       0x63),
     208:  		  moisture("moisture", 		       0x80),
     209:  		  moisture("moisture", 		       0xA0),
     210:  		  moisture("moisture", 		       0xC0),
     211:  		  moisture("moisture", 		       0xE0),

     # internal atmosphere
     301:    RH_and_temp("rh_and_air_temp", "P8_8"),
     302:    RH_and_temp("rh_and_air_temp", "P8_9"),
     303: total_pressure("total_pressure",   2),
     304:         oxygen("O2"),
     305:            CO2("CO2", 5),
     306:            PAR("light", 0xF0),

     # external environment
     401:    RH_and_temp("rh_and_air_temp", "P8_10"),
     402: total_pressure("total_pressure",   1),
     403: 	     PAR("light", 	     0xD0)}

File: API/sensors/sensors.cpp
#include <stdio.h>
#include "sensors.h"

int main()
{
	// liquid tanks and plumbing
	sensor S101 =           EC("S101", "0x66");
	sensor S102 =           pH("S102", "0x65");
	sensor S103 =  temperature("S103", NULL);
	sensor S104 =     DO_probe("S104", "0x61");
	sensor S105 = liquid_level("S105", NULL);
	sensor S106 = liquid_level("S106", NULL);
	sensor S107 = liquid_level("S107", NULL);
	sensor S108 = liquid_level("S108", NULL);
	sensor S109 = liquid_level("S109", NULL);
	sensor S110 =   flow_meter("S110", NULL);
	sensor S111 =   flow_meter("S111", NULL);
	sensor S112 = liquid_level("S112", NULL);
	
	// growth medium
	sensor S201 = temperature("S201", NULL);
	sensor S202 = temperature("S202", NULL);
	sensor S203 = temperature("S203", NULL);
	sensor S204 = temperature("S204", NULL);
	sensor S205 =          EC("S205", NULL);
	sensor S206 =          pH("S206", NULL);
	sensor S208 =    moisture("S208", NULL);
	sensor S209 =    moisture("S209", NULL);
	sensor S210 =    moisture("S210", NULL);
	sensor S211 =    moisture("S211", NULL);
	
	// internal atmosphere
	sensor S301 =        RH_temp("S301", NULL);
	sensor S302 =        RH_temp("S302", "P8_9");
	sensor S303 = total_pressure("S303", "0x77");
	sensor S304 =         oxygen("S304", NULL);
	sensor S305 =            CO2("S305", NULL);
	sensor S306 =      light_PAR("S306", NULL);
	sensor S307 =         camera("S307", NULL);
	
	// external environment
	sensor S401 =        RH_temp("S401", "P8_10");
	sensor S402 = total_pressure("S402", NULL);
	sensor S403 =      light_PAR("S403", NULL);
	
	sensor sensor_suite[32] = {S101, S102, S103, S104, S105, S106, S107, S108, S109, S110, S111, S112, 
				   S201, S202, S203, S204, S205, S206,       S208, S209, S210, S211,
				   S301, S302, S303, S304, S305, S306, S307,
				   S401, S402, S403};
}
